frontend/src/TableComponents/SortMenu.jsx

```diff
@@ -0,0 +1,143 @@
+  ];
+
+  const miscSortOptionsDecreasing = [
+    ["ORB desc", "ORB Descending"],
+    ["DRB desc", "DRB Decreasing"],
+    ["TRB desc", "TRB Decreasing"],
+    ["AST desc", "AST Decreasing"],
+    ["STL desc", "STL Decreasing"],
+    ["BLK desc", "BLK Decreasing"],
+    ["TOV desc", "TOV Decreasing"],
+    ["PF desc", "PF Decreasing"],
+  ];
+
+  const miscSortOptionsIncreasing = [
+    ["ORB desc", "ORB Increasing"],
+    ["DRB desc", "DRB Increasing"],
+    ["TRB desc", "TRB Increasing"],
+    ["AST desc", "AST Increasing"],
+    ["STL desc", "STL Increasing"],
+    ["BLK desc", "BLK Increasing"],
+    ["TOV desc", "TOV Increasing"],
+    ["PF desc", "PF Decreasing"],
+  ];
+
+  const handleClick = (event) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = (event) => {
+    setAnchorEl(null);
+    setSortType(event.target.id);
+  };
+
+  return (
+    <div className="sortMenu">
+      <Button
+        id="sortButton"
+        className="sortButton"
+        aria-controls={open ? "sortMenu" : undefined}
+        aria-haspopup="true"
+        aria-expanded={open ? "true" : undefined}
+        onClick={handleClick}
+      >
+        Sort By
+      </Button>
+      <Menu
+        id="sortMenu"
+        anchorEl={anchorEl}
+        open={open}
+        onClose={handleClose}
+        MenuListProps={{
+          "aria-labelledby": "basic-button",
+        }}
+      >
+        {playerInfoSortOptions.map((option) => (
+          <MenuItem
+            key={option[0]}
+            id={option[0]}
+            className="playerInfoSortOptions"
+            onClick={handleClose}
+          >
+            {option[1]}
+          </MenuItem>
+        ))}
+        <Divider />
+
+        {scoringSortOptionsDecreasing.map((option) => (
+          <MenuItem
+            key={option[0]}
+            id={option[0]}
+            className="scoringSortOptionsDecreasing"
+            onClick={handleClose}
+          >
+            {option[1]}
+          </MenuItem>
+        ))}
+        <Divider />
+
+        {scoringSortOptionsIncreasing.map((option) => (
+          <MenuItem
+            key={option[0]}
+            id={option[0]}
+            className="scoringSortOptionsIncreasing"
+            onClick={handleClose}
+          >
+            {option[1]}
+          </MenuItem>
+        ))}
+        <Divider />
+
+        {miscSortOptionsDecreasing.map((option) => (
+          <MenuItem
+            key={option[0]}
+            id={option[0]}
+            className="miscSortOptionsDecreasing"
+            onClick={handleClose}
+          >
+            {option[1]}
+          </MenuItem>
+        ))}
+        <Divider />
+
+        {miscSortOptionsIncreasing.map((option) => (
+          <MenuItem
+            key={option[0]}
+            id={option[0]}
+            className="miscSortOptionsIncreasing"
+            onClick={handleClose}
+          >
+            {option[1]}
+          </MenuItem>
+        ))}
+      </Menu>
```

Comment:
![Screenshot 2024-07-03 at 4 18 11 PM](https://github.com/GloriaXM/numbers-game/assets/112780860/ad3144bb-ba25-462e-87bb-72b6d0a13eaa)

I repeated the array mapping 4 times because it made sense to split up the sort types into 4 groups for organization/readability sake. Is there a better way to go about this?

Reply:
Consider abstracting this even further by creating another component that takes the array/className as a prop. This new component can locally defined above this component so there's no need to create a new file

Reply:
Since I split the sort bar into two (feature, and sort direction), there was no longer the need to repeat the mapping. But, after doing that, I now had duplicate code for the two sort bars, so I abstracted the sort functionality into one component definition.